rules/signing.bzl

```diff
@@ -3,75 +3,105 @@
 # SPDX-License-Identifier: Apache-2.0
 load("@bazel_skylib//lib:paths.bzl", "paths")
 
-PreSigningBinary = provider(fields = ["files"])
+PreSigningBinaryInfo = provider(fields = ["files"])
 
-def _offline_presigning_artifacts(ctx):
-    digests = []
-    bins = []
-    for src in ctx.files.srcs:
-        pre = ctx.actions.declare_file(paths.replace_extension(src.basename, ".pre-signing"))
-        bins.append(pre)
-        inputs = [
-            src,
-            ctx.file.manifest,
-            ctx.file.rsa_key,
-            ctx.executable._tool,
-        ]
-        spx_args = []
-        if ctx.file.spx_key:
-            spx_args.append("--spx-key={}".format(ctx.file.spx_key.path))
-            inputs.append(ctx.file.spx_key)
-        ctx.actions.run(
-            outputs = [pre],
-            inputs = inputs,
-            arguments = [
-                "--rcfile=",
-                "image",
-                "manifest",
-                "update",
-                "--manifest={}".format(ctx.file.manifest.path),
-                "--rsa-key={}".format(ctx.file.rsa_key.path),
-                "--output={}".format(pre.path),
-                src.path,
-            ] + spx_args,
-            executable = ctx.executable._tool,
-        )
+def _presigning_artifacts(ctx, opentitantool, src, manifest, rsa_key_file, spx_key_file):
+    """Create the pre-signing artifacts for a given input binary.
+
+    Applies he manifest and public components of the keys.  Creates the
+    digests/messages required for signing.
+
+    Args:
+        opentitantool: file; The opentitantool binary.
+        src: file; The source binary
+        manifest: file; The manifest file.
+        rsa_key_file: file; The RSA public key.
+        spx_key_file: file; The SPX+ public key.
+    Returns:
+        (file, file, file): The pre-signing binary, the digest and spx message.
```

Comment:
nit: a mismatch would probably be very easy to catch but I'd prefer a struct or dict return type for clarity.

Reply:
fixed.